unitary/examples/quantum_chinese_chess/move.py

```diff
@@ -15,23 +15,21 @@
 import cirq
 from unitary import alpha
 from unitary.alpha.quantum_effect import QuantumEffect
-from unitary.examples.quantum_chinese_chess.board import Board
 from unitary.examples.quantum_chinese_chess.piece import Piece
 from unitary.examples.quantum_chinese_chess.enums import MoveType, MoveVariant, Type
 
 
 # TODO(): now the class is no longer the base class of all chess moves. Maybe convert this class
 # to a helper class to save each move (with its pop results) in a string form into move history.
-class Move(QuantumEffect):
+class Move:
     """The base class of all chess moves."""
 
     def __init__(
         self,
-        source: str,
-        target: str,
-        board: Board,
-        source2: Optional[str] = None,
-        target2: Optional[str] = None,
+        source: Piece,
```

Comment:
I am not sure if you need a board, but just FYI, if board is only needed for type checking, you can do 

`board: 'Board',` (with quotes)
and add `from __future__ import annotations`
(or hide it behind a `if TYPE_CHECKING` call)

Reply:
Thanks a lot Doug! This is new to me.